Automatically index fk columns on an empty table

rmloveland opened the pull request:
Fixes #4134, #4177.

Summary of changes:

- Update ADD CONSTRAINT doc's section on adding foreign key constraints
  to remove the "you must create an index" language and add some
  "indexes are created for you" language

- Update tables in ADD CONSTRAINT to use new style

- A few small link and copy edits

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/4599)
<!-- Reviewable:end -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/5922b9b8f24ae1e9bc107b463dc5467ef45c1f27/

rmloveland commented:
Direct link to updated section:

- [ADD CONSTRAINT > Add the foreign key constraint with CASCADE](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/5922b9b8f24ae1e9bc107b463dc5467ef45c1f27/v19.1/add-constraint.html#add-the-foreign-key-constraint-with-cascade)

rmloveland commented:
> This looks fine as is, but I wonder if we should be more explicit that adding a foreign key for a non-empty non-indexed table will fail. Do we say that anywhere else?

I think it's implied on the [Foreign key constraint page](https://www.cockroachlabs.com/docs/v19.1/foreign-key.html) where we say among other things that

> Foreign key columns must be indexed

and

> To meet the index requirement when adding the Foreign Key constraint to an existing table, if the columns you want to constraint are not already indexed, use [`CREATE INDEX`](create-index.html) to index them and only then use the [`ADD CONSTRAINT`](add-constraint.html) statement to add the Foreign Key constraint to the columns.

To make it explicit, I've updated this PR in 2 places:

1. Added a small not to the ADD CONSTRAINT page saying 

> Adding a foreign key for a non-empty, non-indexed table will fail, since foreign key columns must be indexed. For more information about the requirements for creating foreign keys, see [Rules for creating foreign keys](foreign-key.html#rules-for-creating-foreign-keys).

2. Updated the 'Foreign Keys' page list of requirements for creating FKs to note that we generate the index for you when you add the FK constraint to an empty table.  Looking at it prompted me to do so.

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ceccea069ce408e6280f77045fc31d03150e2546/

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/bceec9d5c263e3684d5bda68a727b87bfd2d3728/

Edited pages:
- [v19.1/add-constraint.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/bceec9d5c263e3684d5bda68a727b87bfd2d3728/v19.1/add-constraint.html)
- [v19.1/foreign-key.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/bceec9d5c263e3684d5bda68a727b87bfd2d3728/v19.1/foreign-key.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/5d739e929831939bb8039fd12d0810c6d205548e/

Edited pages:
- [v19.1/add-constraint.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/5d739e929831939bb8039fd12d0810c6d205548e/v19.1/add-constraint.html)
- [v19.1/foreign-key.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/5d739e929831939bb8039fd12d0810c6d205548e/v19.1/foreign-key.html)

cockroach-teamcity commented:
Online preview: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3f9f34e380848f56ab3012f2e734064221a4e49d/

Edited pages:
- [v19.1/add-constraint.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3f9f34e380848f56ab3012f2e734064221a4e49d/v19.1/add-constraint.html)
- [v19.1/foreign-key.md](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3f9f34e380848f56ab3012f2e734064221a4e49d/v19.1/foreign-key.html)
